src/Deque.mo

```diff
@@ -33,46 +75,158 @@ module {
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Deque "mo:base/Deque";
+  ///
+  /// let initial = Deque.pushFront(Deque.pushFront(Deque.empty<Nat>(), 2), 1);
+  /// let reduced = Deque.popFront(initial);
+  /// switch reduced {
+  ///     case null assert(false);
+  ///     case (?result) {
+  ///         let removedElement = result.0; // 1
+  ///         let reducedDeque = result.1; // deque with element [2].
+  ///     }
+  /// }
+  /// ```
+  ///
+  /// Runtime: `O(n)` worst-case, amortized to `O(1)`.
+  ///
+  /// Space: `O(n)` worst-case, amortized to `O(1)`.
+  ///
+  /// `n` denotes the number of elements stored in the deque.
+  public func popFront<T>(deque : Deque<T>) : ?(T, Deque<T>) {
+    switch deque {
       case (?(x, f), r) { ?(x, check(f, r)) };
       case (null, ?(x, r)) { ?(x, check(null, r)) };
       case _ { null }
     }
   };
 
-  /// Insert a new element on the back end of the queue
-  public func pushBack<T>(q : Deque<T>, x : T) : Deque<T> {
-    check(q.0, List.push(x, q.1))
+  /// Extend a deque by inserting a new element on the back end.
+  /// Returns a new deque with all the elements of `deque`, followed by `element` on the back.
+  ///
+  /// This may involve dynamic splitting of the internally used two lists.
+  ///
+  /// Example:
+  /// ```
+  /// import Deque "mo:base/Deque";
+  ///
+  /// Deque.pushBack(Deque.pushBack(Deque.empty<Nat>(), 1), 2) // deque with elements [1, 2]
+  /// ```
+  ///
+  /// Runtime: `O(n)` worst-case, amortized to `O(1)`.
+  ///
+  /// Space: `O(n)` worst-case, amortized to `O(1)`.
+  ///
+  /// `n` denotes the number of elements stored in the deque.
+  public func pushBack<T>(deque : Deque<T>, element : T) : Deque<T> {
+    check(deque.0, List.push(element, deque.1))
   };
 
-  /// Inspect the (optional) first element on the back end of the queue
-  public func peekBack<T>(q : Deque<T>) : ?T {
-    switch q {
+  /// Inspect the optional element on the back end of a deque.
+  /// Returns `null` if `deque` is empty. Otherwise, the back element of `deque`.
+  ///
+  /// Example:
+  /// ```motoko
+  /// import Deque "mo:base/Deque";
+  ///
+  /// let deque = Deque.pushBack(Deque.pushBack(Deque.empty<Nat>(), 1), 2);
+  /// Deque.peekBack(deque) // => ?2
+  /// ```
+  ///
+  /// Runtime: `O(1)`.
+  ///
+  /// Space: `O(1)`.
+  ///
+  public func peekBack<T>(deque : Deque<T>) : ?T {
+    switch deque {
       case (f, ?(x, r)) { ?x };
       case (?(x, r), null) { ?x };
       case _ { null }
     }
   };
 
-  /// Remove the first element on the back end of the queue; Returns null when empty.
-  public func popBack<T>(q : Deque<T>) : ?(Deque<T>, T) {
-    switch q {
+  /// Shorten a deque by removing the element on the back end.
+  /// Returns `null` if `deque` is empty. Otherwise, it returns a pair of
+  /// a new deque that contains all the elements of `deque` without the back element
+  /// and as second pair item, the back element.
```

Comment:
```suggestion
  /// Remove an element from the back of the deque.
  /// Returns `null` if `deque` is empty. Otherwise, returns a pair of
  /// the removed element and the remaining elements in a new deque.
```

Similar to the other rephrasing